core/lowering/passes/remove_unnecessary_casts.cpp

```diff
@@ -211,6 +212,149 @@ void RemoveSingleUse0DTensors(std::shared_ptr<torch::jit::Graph>& g) {
   LOG_GRAPH("Post removing single use 0-dim Tensor operations: " << *g);
 }
 
+// Schemas for Aten::Int which can be replaced by scalar equivalents
+const std::unordered_set<c10::Symbol> AtenIntReplacementNodeKinds = {
+    torch::jit::aten::mul,
+    torch::jit::aten::floor_divide,
+};
+
+torch::jit::Value* Validate0DTensor(torch::jit::Value* value) {
+  // Validates that the input Value* is a 0D Tensor (or int/float)
+  // Return the stored int/float Value* if so, otherwise null
+  torch::jit::Value* enclosed_scalar_value = nullptr;
```

Comment:
Personally not a fan of using `nullptr`. There is a type in PyTorch you can use `c10::Optional<torch::jit::Value*>` which is a tagged union that can either contain the value or will be None but avoids the possibility of dereferecing null.